test/spec/services/keywordServiceSpec.js

```diff
@@ -107,4 +107,94 @@ describe('KeywordService', function() {
       expect(result).toEqual(mockData);
     });
   });
+
+  describe('#weightedSearch', function() {
+    var mockData = {
+      'sample-app': {
+        metadata: {
+          name: 'sample-app',
+          annotations: {
+            'openshift.io/display-name': 'Sample App',
+            'openshift.io/description': 'Sample app that uses Maria'
+          }
+        }
+      },
+      'maria-db': {
+        metadata: {
+          name: 'maria-db',
+          annotations: {
+            'openshift.io/display-name': 'Maria DB'
+          }
+        }
+      },
+      'mongo-db': {
+        metadata: {
+          name: 'mongo-db',
+          annotations: {
+            'openshift.io/display-name': 'Mongo DB',
+            tags: 'db'
+          }
+        }
+      }
+    };
+
+    it('should honor field weights', function() {
```

Comment:
So is this the kind of configuration we might want to do once up front, via `$provider` in `app.js` `.cofig()`, or would we want to custom config weights & paths for every use?

Reply:
ie, `config` some functions & cache them, so we don't have copy / paste repeats of the setup in multiple places.  I don't know how  many times we will use this search feature.